crates/cli/build.rs

```diff
@@ -0,0 +1,46 @@
+use std::{env, error::Error};
+
+use vergen::EmitBuilder;
+
+#[allow(clippy::disallowed_macros)]
+fn main() -> Result<(), Box<dyn Error>> {
```

Comment:
could we have a `FoundryVersion` to be reused in both places? (smth like `FoundryVersion.with_nightly_tag`, `.with_build_timestamp`)

Reply:
Turns out we didn't need the `build_timestamp` anymore making it more practical to unify this in a single `build.rs` and `lib.rs` (for exports) in `foundry_common` that is then used in the different places